torch/_library/infer_schema.py

```diff
@@ -64,7 +66,10 @@ def convert_type_string(annotation_type: str):
             )
 
         schema_type = SUPPORTED_PARAM_TYPES[annotation_type]
-        if name in mutates_args:
+        if mutates_args == UNKNOWN_MUTATES:
+            if schema_type.startswith("Tensor"):
+                schema_type = f"Tensor(a{idx}!){schema_type[len('Tensor'):]}"
```

Comment:
let's assert that if mutates_args is a string, then it must be UNKNOWN_MUTATES. Otherwise you can actually iterate through a string.